src/compute-types/src/time_dependence.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Description of how a dataflow follows time, independent of time.
+
+use mz_proto::{RustType, TryFromProtoError};
+use mz_repr::refresh_schedule::RefreshSchedule;
+use mz_repr::Timestamp;
+use proptest::arbitrary::{any, Arbitrary};
+use proptest::prelude::BoxedStrategy;
+use proptest::strategy::{Just, Strategy, Union};
+use serde::{Deserialize, Serialize};
+
+include!(concat!(
+    env!("OUT_DIR"),
+    "/mz_compute_types.time_dependence.rs"
+));
+
+/// Description of how a dataflow follows time.
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, Ord, PartialOrd)]
+pub enum TimeDependence {
```

Comment:
Today we already have both `enum Timeline` and `enum TimelineContext`, this PR introduces `enum TimeDependence` which makes this space feel a bit more crowded. Totally understand this is probably the best way to model the space, but do you mind adding to this doc comment how all the types relate, specifically how this type differs from `Timeline`? Also maybe in `dataflow_expiration.rs` explaining what it means for a dataflow to "expire"?